dotnet/src/Connectors/Connectors.Memory.AzureCosmosDBNoSQL/AzureCosmosDBNoSQLVectorStoreRecordCollection.cs

```diff
@@ -0,0 +1,529 @@
+
+        this._partitionKeyStoragePropertyName = this._storagePropertyNames[partitionKeyPropertyName];
+
+        this._nonVectorStoragePropertyNames = properties.DataProperties
+            .Cast<VectorStoreRecordProperty>()
+            .Concat([properties.KeyProperty])
+            .Select(x => this._storagePropertyNames[x.DataModelPropertyName])
+            .ToList();
+    }
+
+    /// <inheritdoc />
+    public Task<bool> CollectionExistsAsync(CancellationToken cancellationToken = default)
+    {
+        return this.RunOperationAsync("CollectionExists", async () =>
+        {
+            const string Query = "SELECT VALUE(c.id) FROM c WHERE c.id = @collectionName";
+
+            var queryDefinition = new QueryDefinition(Query).WithParameter("@collectionName", this.CollectionName);
+
+            using var feedIterator = this._database.GetContainerQueryIterator<string>(queryDefinition);
+
+            while (feedIterator.HasMoreResults)
+            {
+                var next = await feedIterator.ReadNextAsync(cancellationToken).ConfigureAwait(false);
+
+                foreach (var containerName in next.Resource)
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        });
+    }
+
+    /// <inheritdoc />
+    public Task CreateCollectionAsync(CancellationToken cancellationToken = default)
+    {
+        return this.RunOperationAsync("CreateContainer", () =>
+            this._database.CreateContainerAsync(this.GetContainerProperties(), cancellationToken: cancellationToken));
+    }
+
+    /// <inheritdoc />
+    public async Task CreateCollectionIfNotExistsAsync(CancellationToken cancellationToken = default)
+    {
+        if (!await this.CollectionExistsAsync(cancellationToken).ConfigureAwait(false))
+        {
+            await this.CreateCollectionAsync(cancellationToken).ConfigureAwait(false);
+        }
+    }
+
+    /// <inheritdoc />
+    public Task DeleteAsync(string key, DeleteRecordOptions? options = null, CancellationToken cancellationToken = default)
+    {
+        Verify.NotNullOrWhiteSpace(key);
+
+        return this.DeleteItemsAsync([key], cancellationToken);
+    }
+
+    /// <inheritdoc />
+    public Task DeleteBatchAsync(IEnumerable<string> keys, DeleteRecordOptions? options = null, CancellationToken cancellationToken = default)
+    {
+        Verify.NotNull(keys);
+
+        return this.DeleteItemsAsync(keys, cancellationToken);
+    }
+
+    /// <inheritdoc />
+    public Task DeleteCollectionAsync(CancellationToken cancellationToken = default)
+    {
+        return this.RunOperationAsync("DeleteContainer", () =>
+            this._database
+                .GetContainer(this.CollectionName)
+                .DeleteContainerAsync(cancellationToken: cancellationToken));
+    }
+
+    /// <inheritdoc />
+    public async Task<TRecord?> GetAsync(string key, GetRecordOptions? options = null, CancellationToken cancellationToken = default)
```

Comment:
I wonder if it would make sense to have an additional interface implementation that uses a composite key that contains both PartitionKey and Key.  Not for this PR, but something to consider.

Reply:
Great idea, which also fits nicely into current abstraction. I implemented additional interface with composite key.